File: trunk/src/core/src/main/java/org/semachina/jena/datatype/factory/DurationFactory.java
package org.semachina.jena.datatype.factory;

import com.hp.hpl.jena.vocabulary.XSD;
import org.joda.time.Period;
import org.joda.time.format.ISOPeriodFormat;
import org.semachina.jena.impl.SimpleRDFDatatypeImpl;

public class DurationFactory extends SimpleRDFDatatypeImpl<Period> {
    public DurationFactory() {
        super(XSD.duration.getURI(), Period.class);
    }

    @Override
    public Period parseLexicalForm(String lexicalForm) throws Exception {
        return ISOPeriodFormat.standard().parsePeriod(lexicalForm);
    }

    @Override
    public String toLexicalForm(Period cast) throws Exception {
        return ISOPeriodFormat.standard().print(cast);
    }
}

File: trunk/src/core/src/main/scala/org/semachina/jena/datatype/factory/DurationFactory.scala
package org.semachina.jena.datatype.factory

import com.hp.hpl.jena.vocabulary.XSD
import org.joda.time.Period
import org.joda.time.format.ISOPeriodFormat
import org.semachina.jena.datatype.SimpleRDFDatatype

class DurationFactory extends SimpleRDFDatatype[Period] (
  XSD.duration.getURI,
  { lexicalForm: String => ISOPeriodFormat.standard.parsePeriod(lexicalForm) },
  { cast: Period => ISOPeriodFormat.standard.print(cast) } )